src/trusted-urls.js

```diff
@@ -0,0 +1,29 @@
+const { config } = require('./config')
+
+exports.checkUrlUnsafe = urlStr => {
+  if (!config.urlSafeList) {
+    return null
+  } else {
+    let url
+    try {
+      // example.com is an owned domain known to not be malicious
+      const base = 'https://example.com'
+
+      url = new URL(urlStr, base);
+    } catch (err) {
+      return `Invalid URL - ${urlStr}`
+    }
+
+    const isSafe = config.urlSafeList.some(safe => {
+      if (typeof safe === 'function') {
+        return safe(url)
```

Comment:
potentially wrap this in try/catch just in case the supplied function is weird? that way the server doesn't die here but could just return null or something.

Reply:
express is actually smart enough to not kill the server when this happens. It sends an HTTP 500 with the stacktrace as an HTML page.

Reply:
Example express response:

```
http PATCH :8000/import-map.json imports:='{"thing": "https://unpkg.com/browse/react@16.13.0/index.js"}'
HTTP/1.1 500 Internal Server Error
Connection: keep-alive
Content-Length: 1170
Content-Security-Policy: default-src 'none'
Content-Type: text/html; charset=utf-8
Date: Fri, 28 Feb 2020 22:48:48 GMT
X-Content-Type-Options: nosniff
X-Powered-By: Express

<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Error<br> &nbsp; &nbsp;at module.exports.urlSafeList (/Users/joeldenning/code/import-map-deployer/conf.js:6:13)<br> &nbsp; &nbsp;at /Users/joeldenning/code/import-map-deployer/src/trusted-urls.js:19:16<br> &nbsp; &nbsp;at Array.some (&lt;anonymous&gt;)<br> &nbsp; &nbsp;at exports.checkUrlUnsafe (/Users/joeldenning/code/import-map-deployer/src/trusted-urls.js:17:39)<br> &nbsp; &nbsp;at Array.map (&lt;anonymous&gt;)<br> &nbsp; &nbsp;at /Users/joeldenning/code/import-map-deployer/src/web-server.js:128:33<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/Users/joeldenning/code/import-map-deployer/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at next (/Users/joeldenning/code/import-map-deployer/node_modules/express/lib/router/route.js:137:13)<br> &nbsp; &nbsp;at Route.dispatch (/Users/joeldenning/code/import-map-deployer/node_modules/express/lib/router/route.js:112:3)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/Users/joeldenning/code/import-map-deployer/node_modules/express/lib/router/layer.js:95:5)</pre>
</body>
</html>
```